chart/templates/deployment.yaml

```
@@ -61,4 +66,12 @@ spec:
         - name: {{ $name }}
         {{- tpl (toYaml $config) $ | nindent 10 }}
         {{- end }}
+        {{- if $.Values.customNetworkConfig }}
+        - name: custom-network-config-volume
+          configMap:
+            name: custom-networks-config
+            items:
+              - key: networks-config.json
+                path: networks-config.json
```

Shouldn't need to explicitly list all files. Also would be good to set more secure permissions.
```suggestion
            defaultMode: 420
```